test/dynamo/test_misc.py

```diff
@@ -3069,6 +3070,51 @@ def guard_failures(failure):
                 "tensor 'x' size mismatch at index 0. expected 2, actual 3",
             )
 
+    def test_restore_graphstate(self):
```

Comment:
I think you can write this test without dipping into internals. What you need is to write some code that makes dynamo: checkpoint, add a guard, and then trigger a rollback. Then you need to test the guard does not exist (but we have an API for that.) Is that enough to go on?

Reply:
The lack of a guard is not a good test - It does not prove rollback worked because there is no useful counterfactual. I can do that, if you like, but it feels weak.

Reply:
I'm generally not a fan of test cases that touch internals, because they're a pain to update when said internals get refactored. XD It is true that it can be sometimes be difficult to tell if the synthesized test case actually did the thing you want it to; e.g., in this case, I think what you are worried about is showing that a guard would have actually be installed had we not rolled back, is that right? But that's also a strength: the test will keep working throughout all refactors of the code, and you can think of it as just a test that is triggering some tricky code.

I think I'm OK with having both tests, with the understanding that if the internal test becomes annoying to update, it's ok for the future dev to delete it. Best to have a comment saying so.

Reply:
Agreed on all fronts, will comment